test/Interop/Cxx/union/Inputs/anonymous-union-partly-invalid.h

```diff
@@ -0,0 +1,16 @@
+@class C;
+@interface C
+{}
+@end
+
+struct S {
+  union {
+    C *t;
```

Comment:
[From the other patch:](https://github.com/apple/swift/pull/37265/files#r626255086)
> I believe because it thinks the destructor is deleted at
> 
> https://github.com/apple/swift/blob/13449004a069f7d39d841c675c07d7d6debf0f4a/lib/ClangImporter/ImportDecl.cpp#L3352
> 
> in `isCxxRecordImportable`

I'm a bit confused. Where is the destructor deleted? 

If that's the problem I think we actually need to fix this differently: figure out why clang thinks the destructor is deleted here (or maybe I'm just missing something 😁). Then we should try to reproduce this assertion again using a "differently invalid" type.

Reply:
Actually I think it's because the destructor for an anonymous union here is not deleted (sorry read the condition inverted), and since I dont think theres no way to do a `~() = delete;` I can't try this out otherwise for an anonymous union. I think the reason for this is that in this similar instance:

```
@interface C
@end

union {
  C *t;
  char c;
};

```

The anon union contains an arc managed pointer so the destructor can't be trivial but I am not certain of this.

Reply:
If the destructor is not deleted, then why can't we import it? I want to make sure we completely understand the problem here.

Reply:
Here's an idea about why this might be failing, it comes from the observation that a similar C++ program doesn't fail:

```
struct C {
  C(const C&) = delete;
};

struct S {
  union {
    OC *t;
    char c;
  };
  S(const S &s) {}
  ~S() { }
  int f() { return 42; }
};

S *getSPtr();
```
In the above program, `C` isn't able to be imported, but its pointer type is still imported as `OpaquePointer` so we get:

```
struct S {
  struct __Unnamed_union___Anonymous_field0 {
    var t: OpaquePointer!
    var c: CChar
    init()
    init(t: OpaquePointer!)
    init(c: CChar)
  }
  var __Anonymous_field0: S.__Unnamed_union___Anonymous_field0
  var t: OpaquePointer!
  var c: CChar
  mutating func f() -> Int32
}
func getSPtr() -> UnsafeMutablePointer<S>!
```
However, as I understand it (and this might not be correct), this is not the case for ObjC types: if they can't be imported, their type is nullptr. This might be the reason it's only happening for ObjC. 

However, I'm still not sure we know the answer to the question: why is `isCxxRecordImportable` returning `false`?